extensions/deepcast/src/utils.ts

```diff
@@ -5,21 +5,45 @@ interface Preferences {
   key: string;
   pro: boolean;
 }
-const translate = async (target: string) => {
-  const preferences: Preferences = getPreferenceValues();
-  const key = preferences.key;
-  const pro = preferences.pro;
+
+export function getPreferences() {
+  return getPreferenceValues<Preferences>();
+}
+
+export async function sendTranslateRequest({
+  text: initialText,
+  sourceLanguage,
+  targetLanguage,
+}: {
+  text?: string;
+  sourceLanguage?: string;
+  targetLanguage: string;
+}) {
   try {
-    const text = await getSelectedText();
+    const text = initialText || (await getSelectedText());
+
+    const { key, pro } = getPreferences();
+
     try {
-      const response = await got(
-        `https://api${pro ? "" : "-free"}.deepl.com/v2/translate?auth_key=${key}&text=${text}&target_lang=${target}`
-      );
-      const translation = JSON.parse(response.body).translations[0].text;
+      const {
```

Comment:
If we do a rewrite of that logic, could we use https://developers.raycast.com/utilities/react-hooks/usefetch or classic fetch?

Reply:
`useFetch` doesn't seem to be an option, as this function is called as a command as well (not in React component).

`fetch` also requires a third party library (e.g. `node-fetch`), as global fetch was deprecated (see [Raycast Changelog 1.46.0 - 2023-01-18](https://developers.raycast.com/changelog#1.46.0-2023-01-18))

Reply:
Ahh didn't saw that Raycast deprecated the experimental fetch in Node. We can switch to it after it's stable release.